### PR TITLE
Make Vimeo request() generic to optionally type results

### PR DESCRIPTION
**Pull request by mattleff**
Please fill in this template.

- [x] Use a meaningful title for the pull request. Include the name of the package modified.
- [x] Test the change in your own code. (Compile and run.)
- [x] Add or edit tests to reflect the change. (Run with `npm test`.)
- [x] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#make-a-pull-request).
- [x] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#common-mistakes).
- [x] Run `npm run lint package-name` (or `tsc` if no `tslint.json` is present).

If changing an existing definition:
- [ ] Provide a URL to documentation or source code which provides context for the suggested changes: <<url here>>
- [ ] If this PR brings the type definitions up to date with a new version of the JS library, update the version number in the header.
- [x] Include [tests for your changes](https://github.com/DefinitelyTyped/DefinitelyTyped#testing)
- [x] If you are making substantial changes, consider adding a `tslint.json` containing `{ "extends": "dtslint/dt.json" }`. If for reason the any rule need to be disabled, disable it for that line using `// tslint:disable-next-line [ruleName]` and not for whole package so that the need for disabling can be reviewed.

### CODE REVIEW

**Comment by andrewbranch** [comment]
ping @typescript-bot, what’s with the labels

**Comment by mattleff** [comment]
@andrewbranch Thanks for fixing the dt-mergebot issues. Is this PR good to go now?

File: types/vimeo/index.d.ts [hunk 1]
```diff
@@ -59,7 +59,8 @@ export class Vimeo {
      *                                  `host`, `port`, `query` or `headers`.
      * @param callback  Called when complete, `function (err, json)`.
      */
-    request(url: string | RequestOptions, callback: CompleteCallback): void;
+    // tslint:disable-next-line no-unnecessary-generics
+    request<T>(url: string | RequestOptions, callback: CompleteCallback<T>): void;
```
Comment (andrewbranch) [inline types/vimeo/index.d.ts:63]: The fact that you have to disable the lint rule tells you that this is a bit of an antipattern. Using uninferrable generics is just a trickier way of writing an unsafe downcast. For the very TypeScript-savvy, it’s a more compact syntax, but for folks who are coming to TypeScript from languages with reflection like C# or Java, this pattern looks dangerously like it _does something_ when it doesn’t. I am pretty strongly against this change (and all no-unnecessary-generics exemptions), but won’t stop you outright as the sole author of these definitions—instead I’ll just hope I can convince you that this has the potential to do more harm than good.

Comment (mattleff) [reply types/vimeo/index.d.ts]: Yeah, that makes sense. I've removed the generic and just typed the callback response `any` (8e49bd4) and I'll assert the type at the call sites.

**Review by andrewbranch — APPROVED** [review]
Thanks!

**Comment by mattleff** [comment]
Ready to merge

**Pull request merged**